sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -100,10 +111,26 @@ class Iso8601TimestampFormatter(
  */
 class FractionTimestampFormatter(zoneId: ZoneId)
   extends Iso8601TimestampFormatter(
-    "", zoneId, TimestampFormatter.defaultLocale, needVarLengthSecondFraction = false) {
+    TimestampFormatter.defaultPattern,
+    zoneId,
+    TimestampFormatter.defaultLocale,
+    needVarLengthSecondFraction = false) {
 
   @transient
   override protected lazy val formatter = DateTimeFormatterHelper.fractionFormatter
+
+  // Converts Timestamp to string according to Hive TimestampWritable convention.
+  // The code is borrowed from Spark 2.4 DateTimeUtils.timestampToString
```

Comment:
is it really needed? `DateTimeFormatterHelper.fractionFormatter` should omit tailing 0 already.

Reply:
> should omit tailing 0 already.

The reason of making this PR is to pass java.sql.Timestamp to a legacy formatter which can accept the type but the legacy formatter of fractionFormatter is SimpleDateFormat **which cannot omit tailing 0.**

> is it really needed?

I think so. What do you propose instead of it?

Reply:
Let's make the comment more clear:
```
// The new formatter will omit the trailing 0 in the timestamp string, but the legacy formatter can't.
// Here we borrow the code from Spark 2.4 DateTimeUtils.timestampToString to omit the
// trailing 0 for the legacy formatter as well.
```
We don't need to mention hive at all. This is just for internal consistency.

Reply:
Thanks. I updated the comment.